### DISCUSSION TITLE
Increase Function Payload Limit

### DISCUSSION THREAD

**Discussion by agonsalves** [gatsbyjs/gatsby#35084] [Ideas / Feature Requests] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

When creating an API/handler through Gatsby Functions, the handler is limited to requests under 100KB in size, as it uses the default express.js limit. This limit cannot be changed on the dev server, but it should be editable.

### Reproduction Link

https://github.com/agonsalves/gatsby-bug-repro

### Steps to Reproduce

1. Attach a file larger than 100KB to the field.
2. Hit submit
3. Look in your server console for the error: PayloadTooLargeError: request entity too large
...

### Expected Result

The request shouldn't result in an error - as long as the request limit was high enough. However, there is no way to change the limit on the development server.

### Actual Result

ERROR: PayloadTooLargeError: request entity too large

### Environment

```shell
System:
    OS: macOS 12.1
    CPU: (10) arm64 Apple M1 Pro
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.14.0 - /opt/homebrew/bin/node
    Yarn: 1.22.17 - /opt/homebrew/bin/yarn
    npm: 8.3.1 - /opt/homebrew/bin/npm
  Languages:
    Python: 2.7.18 - /usr/bin/python
  Browsers:
    Chrome: 99.0.4844.51
    Safari: 15.2
  npmPackages:
    gatsby: ^4.9.2 => 4.9.2 
  npmGlobalPackages:
    gatsby: 4.9.1
```

**Comment by LekoArts** [comment]
cc @EzraEllette as you said you'll be working on this

**Comment by LekoArts** [comment]
@EzraEllette is working on this at the moment

**Comment by SukantaB** [comment, marked as answer]
Hi, @LekoArts  Do we have any expected release timeline for this ?

**Reply by pieh** [reply]
Hey @SukantaB (and @agonsalves ).

We are currently working on providing way to adjust default config - you can follow it in https://github.com/gatsbyjs/gatsby/pull/35393

To fully ship this feature we need similar handling on hosting side of things as well as production hosting won't use our `gatsby serve` and rather will have specialized infra running generated functions - this is just signalling that it's not enough to just make changes in `gatsby` package

**Reply by pieh** [reply]
`gatsby@4.14` was published with ability to bump the limit - see https://www.gatsbyjs.com/docs/reference/functions/middleware-and-helpers/#custom-body-parsing for details and examples
